opentelemetry-sdk/tests/trace/test_sampling.py

```diff
@@ -329,11 +331,101 @@ def exec_parent_based(self, parent_sampling_context):
             self.assertEqual(not_sampled_result.attributes, {})
             self.assertEqual(not_sampled_result.trace_state, trace_state)
 
+        with parent_sampling_context(
```

Comment:
nit: these tests could be refactored to make them easier, there's a lot of duplicate code here